### DISCUSSION TITLE
Request: add report lineage (including page... and vis?)

### DISCUSSION THREAD

**Discussion by Dalcastle** [migueesc123/PowerBIRESTAPI#186] [Ideas]
First, I LOVE THIS TOOL! Thanks for building and sharing it! One thing I've been chasing for a long time now is full lineage... ideally datasource/set, report, page, vis, columns/measures with the filters relevant to each level of the report. I know this is challenging as I've been trying for over a year to get it but this makes report documentation automatic!

DMV's have lists of measures in data sources but can't tell where used. PBIX files have the measures used and filters but too hard to get robustly...

Anyway, would be really great! (probably worth putting behind a pay wall...)

**Comment by BeSmarterWithData** [comment]
@Dalcastle I've just discovered this repository so I'm still exploring use cases....but based on your comment, you'd likely benefit from a repository I posted a month or so ago. 

https://github.com/chris1642/PBIVIsualObjectLayer

In short, it's a method to automatically download, extract, and combine the Visual Object Layer of all reports within a workspace, into 1 file (so you can see exactly what and where measures and fields are used in reports).

We use a version of it to combine our dataset, visual reports, and usage detail.
